File: Pi.lua
function calculatePiLeibniz(iterations)
    local pi = 0
    for i = 0, iterations - 1 do
        local term = 1 / (2 * i + 1)
        if i % 2 == 0 then
            pi = pi + term
        else
            pi = pi - term
        end
    end
    return pi * 4
end

print(calculatePiLeibniz(1000000))      

File: Pi_in_Kotlin/Pi.kt
import kotlin.math.PI

fun main() {
    // 1. Using the built-in constant (Recommended)
    val radius = 5.0
    val circumference = 2 * PI * radius
    println("Circumference: $circumference")

    // 2. Computing Pi using Leibniz formula (Approximation)
    val computedPi = calculatePiLeibniz(1000000)
    println("Computed Pi: $computedPi")
}

fun calculatePiLeibniz(iterations: Int): Double {
    var pi = 0.0
    var denominator = 1
    for (index in 0 until iterations) {
        if (index % 2 == 0) {
            pi += 4.0 / denominator
        } else {
            pi -= 4.0 / denominator
        }
        denominator += 2
    }
    return pi
}
